### PR TITLE
LCOW: Spot fix for multi-os image pulling

### PR DESCRIPTION
**Pull request by lowenna**
Signed-off-by: John Howard <jhoward@microsoft.com>

@johnstep @tianon @friism As per slack discussion, quick spot fix for multi-arch images pulling in LCOW. Verified busybox and Ubuntu pull just fine with the multi-arch manifests now - eg

```
PS C:\lcow2> docker pull ubuntu
Using default tag: latest
latest: Pulling from library/ubuntu
d5c6f90da05d: Pull complete
1300883d87d5: Pull complete
c220aa3cfc1b: Pull complete
2e9398f099dc: Pull complete
dc27a084064f: Pull complete
Digest: sha256:9c24600e5c4b09a51c933e1a1bd4748c2e4a639787421132db8f42fb3448d66b
Status: Downloaded newer image for ubuntu:latest
PS C:\lcow2>
```

### CODE REVIEW

**Comment by tianon** [comment]
As another test, `hello-world:latest` should have both Windows and Linux in a single manifest list. 🤘

**Comment by jhowardmsft** [comment]
@Tianon - this good enough? 😺 

```
PS C:\lcow2> docker pull hello-world:latest
latest: Pulling from library/hello-world
5b0f327be733: Pull complete
Digest: sha256:88b1272b6f6131d46a53fcc2bb0ac836467f38a05916587cbb483f3dd66f3371
Status: Downloaded newer image for hello-world:latest
PS C:\lcow2> docker run hello-world:latest

Hello from Docker!
This message shows that your installation appears to be working correctly.

To generate this message, Docker took the following steps:
 1. The Docker client contacted the Docker daemon.
 2. The Docker daemon pulled the "hello-world" image from the Docker Hub.
 3. The Docker daemon created a new container from that image which runs the
    executable that produces the output you are currently reading.
 4. The Docker daemon streamed that output to the Docker client, which sent it
    to your terminal.

To try something more ambitious, you can run an Ubuntu container with:
 $ docker run -it ubuntu bash

Share images, automate workflows, and more with a free Docker ID:
 https://cloud.docker.com/

For more examples and ideas, visit:
 https://docs.docker.com/engine/userguide/

PS C:\lcow2>
```

**Comment by jhowardmsft** [comment]
And for completeness, in non-lcow mode

PS C:\lcow2> docker pull hello-world:latest
latest: Pulling from library/hello-world
bce2fbc256ea: Pull complete
38cc73423ca1: Pull complete
1628e02ffaf1: Pull complete
cfeaed3be9bf: Pull complete
Digest: sha256:88b1272b6f6131d46a53fcc2bb0ac836467f38a05916587cbb483f3dd66f3371
Status: Downloaded newer image for hello-world:latest
PS C:\lcow2> docker run --rm --isolation=hyperv hello-world:latest

Hello from Docker!
This message shows that your installation appears to be working correctly.

To generate this message, Docker took the following steps:
 1. The Docker client contacted the Docker daemon.
 2. The Docker daemon pulled the "hello-world" image from the Docker Hub.
 3. The Docker daemon created a new container from that image which runs the
    executable that produces the output you are currently reading.
 4. The Docker daemon streamed that output to the Docker client, which sent it
    to your terminal.

To try something more ambitious, you can run a Nano Server container with:
 $ docker run -it microsoft/nanoserver powershell

Share images, automate workflows, and more with a free Docker ID:
 https://cloud.docker.com/

For more examples and ideas, visit:
 https://docs.docker.com/engine/userguide/

**Review by johnstep — APPROVED** [review]
LGTM (verified before and after)

**Comment by tiborvass** [comment]
LGTM

**Comment by tianon** [comment]
Neato! 👍😃

**Comment by jhowardmsft** [comment]
@Tianon BTW there's a bug in the Windows hello-world:

`$ docker run -it microsoft/nanoserver powershell`

Should be `PS C:\> docker run -it microsoft/nanoserver powershell`

**Comment by jhowardmsft** [comment]
Oh, and maybe not `powershell` as the target exe based on other conversations.....

**Comment by tianon** [comment]
🔥🔥🔥🔥🔥🔥🔥

(Thanks for filing bugs 👍)

**Comment by jhowardmsft** [comment]
@Tianon - z failure looks legit. Not had time to investigate. Although it's better than current master where all CI fails. 

```
22:33:21 ----------------------------------------------------------------------
22:33:21 FAIL: docker_cli_pull_test.go:194: DockerHubPullSuite.TestPullAllTagsFromCentralRegistry
22:33:21 
22:33:21 docker_cli_pull_test.go:201:
22:33:21     s.Cmd(c, "pull", "--all-tags=true", "busybox")
22:33:21 docker_hub_pull_suite_test.go:74:
22:33:21     c.Assert(err, checker.IsNil, check.Commentf("%q failed with errors: %s, %v", strings.Join(arg, " "), out, err))
22:33:21 ... value *exec.ExitError = &exec.ExitError{ProcessState:(*os.ProcessState)(0xc4201ca2c0), Stderr:[]uint8(nil)} ("exit status 1")
22:33:21 ... "--all-tags=true busybox" failed with errors: 1-glibc: Pulling from library/busybox
22:33:21 Digest: sha256:748f644f30eea219bfe4476e23ec6a62b34aaa90e25ce069c810ee6bcdf183ec
22:33:21 1-musl: Pulling from library/busybox
22:33:21 0315c83cf751: Pulling fs layer
22:33:21 0315c83cf751: Verifying Checksum
22:33:21 0315c83cf751: Download complete
22:33:21 0315c83cf751: Pull complete
22:33:21 Digest: sha256:01095b6c49cd4d1e388e70a791f0b5213dc430ccad750f7ea47348524c6e4e73
22:33:21 1-ubuntu: Pulling from library/busybox
22:33:21 082340653daf: Pulling fs layer
22:33:21 a3ed95caeb02: Pulling fs layer
22:33:21 a3ed95caeb02: Verifying Checksum
22:33:21 a3ed95caeb02: Download complete
22:33:21 082340653daf: Verifying Checksum
22:33:21 082340653daf: Download complete
22:33:21 082340653daf: Pull complete
22:33:21 a3ed95caeb02: Pull complete
22:33:21 Digest: sha256:148a142e0eb29d0c686e27dc08e399496cf46ca64ed6303839c339ed8dabd3ce
22:33:21 1-uclibc: Pulling from library/busybox
22:33:21 no matching manifest for linux/s390x in the manifest list entries
22:33:21 , exit status 1
22:33:21 
22:33:39
```

**Comment by jhowardmsft** [comment]
PPC failure same as Z.

**Comment by jhowardmsft** [comment]
The PPC/Z failure is unrelated - needs fixing due to the change of busybox to multi-arch. Shouldn't block this PR which fixes RS1/janky/experimental which are all failing on master otherwise.

**Comment by jhowardmsft** [comment]
Think I have the z/ppc fix. Pushed update. Not going to wait up 2 hours to verify though....

**Comment by jhowardmsft** [comment]
@vdemeester @thaJeztah @yongtang In case it's green come 1AM PST. Please merge :) Thx!

**Comment by thaJeztah** [comment]
No luck yet for power/z:

```
08:22:52 no matching manifest for linux/ppc64le in the manifest list entries
```

```
08:26:40 no matching manifest for linux/s390x in the manifest list entries
```

I restarted Janky, because if failed for unapparent reason

**Comment by tophj-ibm** [comment]
this looks like a bug with '--all-tags=true', because the busybox manifest list definitely has p/z manifests and I can easily recreate this.

**Comment by thaJeztah** [comment]
@tophj-ibm thanks for looking! 

I see Janky now passed: Given that this PR resolves the other failures, I'll merge this one. Would you be able to have a look at the failing p/z and open a follow-up?

**Comment by tophj-ibm** [comment]
@thaJeztah yep, I'm looking at it now

**Comment by thaJeztah** [comment]
Thanks! <3

**Comment by jhowardmsft** [comment]
I see what I did wrong - I left a .Not in the else condition of the test. That would have made ppc and z pass, but probably artificially, as the image should have pulled successfully. I think. (Only looking from phone so may be missing something else)

**Comment by thaJeztah** [comment]
I decided to just merge as-is for now, but if you have thoughts on a solution, happy to hear of course :+1:

**Comment by tophj-ibm** [comment]
no that also doesn't work, I was wrong. That specific image:tag doesn't have a p/z equivalent so the pull itself errors out when it reaches it. 

Checking for an error could possibly work and I'll try it. I'm currently testing different, smaller repos as well because busybox takes a long time even if it passes (look at all of these tags https://hub.docker.com/r/library/busybox/tags/ )

**Comment by tophj-ibm** [comment]
I think I figured out what you were trying to do @jhowardmsft, opened pr: #34837

**Pull request merged**
